front/src/modules/settings/profile/components/DeleteWorkspace.tsx

```diff
@@ -0,0 +1,128 @@
+import { Dispatch, FC, SetStateAction, useEffect, useState } from 'react';
+import { useTheme } from '@emotion/react';
+import styled from '@emotion/styled';
+import debounce from 'lodash.debounce';
+import { useRecoilValue } from 'recoil';
+
+import { StyledContainer } from '@/auth/components/Modal';
+import { currentUserState } from '@/auth/states/currentUserState';
+import { Button } from '@/ui/button/components/Button';
+import { TextInput } from '@/ui/input/components/TextInput';
+import { Modal } from '@/ui/modal/components/Modal';
+import { SubSectionTitle } from '@/ui/title/components/SubSectionTitle';
+
+const CenteredButton = styled(Button)`
+  align-items: center;
+  display: flex;
+  justify-content: center;
+`;
+
+const StyledDeleteButton = styled(CenteredButton)`
+  border-color: ${({ theme }) => theme.color['red20-1']};
+  border-width: '1px';
+  color: ${({ theme }) => theme.color.red};
+  font-size: '13px';
+  font-weight: 500;
+  line-height: '150%';
+`;
+
+const StyledModalContainer = styled(StyledContainer)`
+  color: ${({ theme }) => theme.font.color.primary};
+  > * + * {
+    margin-top: ${({ theme }) => theme.spacing(8)};
+  }
+`;
+
+const EmailField: FC<{
+  value: string;
+  setValue: Dispatch<SetStateAction<string>>;
+  userEmail?: string;
+  isDeleteDisabled: boolean;
+}> = ({ value, setValue, userEmail, isDeleteDisabled }) => {
+  const errorMessage =
+    value && isDeleteDisabled ? 'email provided is not correct' : '';
+
+  return (
+    <TextInput
+      value={value}
+      placeholder={userEmail}
+      fullWidth
+      key={'email-' + userEmail}
+      onChange={setValue}
+      error={errorMessage}
+    />
+  );
+};
+
+export const DeleteWorkspace = () => {
```

Comment:
We don't use arrowFunction to defined components neither, please use export function DeleteWorkspace() {} syntax